### PR TITLE
Add support for MemoryCacheEntryOptions.ExpirationTokens

### PR DESCRIPTION
**Pull request by b-twis**
Add support to FusionCacheEntryOptions for ExpirationTokens.

Update FusionCache with Memory.Eviction event to call Remove when a Token is expired. This forces complete eviction of cache entries out of distributed and sending backplane notifications

Added tests for Memory Cache eviction.

Update Docs for Options

Addresses https://github.com/ZiggyCreatures/FusionCache/discussions/61

### CODE REVIEW

**Comment by jodydonetti** [comment]
Hi @b-twis , thanks for the PR!
I will take a look at it as soon as possible.

**Comment by b-twis** [comment]
@jodydonetti sorry, had some merge issues getting it aligned after the most recent changes which required a force-push to cleanup. 

I have reapplied the changes onto the latest main branch, if you could please take a look.

**Comment by jodydonetti** [comment]
Hi @b-twis , first of all thanks for your contribution!

Before moving further, I've trid to answer your design considerations in the original issue [here](https://github.com/ZiggyCreatures/FusionCache/discussions/61#discussioncomment-5159731).

Let me know what you think.

File: docs/Options.md
Comment (tacosontitan) [inline docs/Options.md:1]: Other entries utilize back ticks to render code related names such as `CancellationTokenSource` and `CancellationChangeToken` in `<code>` tags here on GitHub. I don't see where it hurts to *not* utilize them, but it breaks consistency with the rest of the file by doing so.

File: src/ZiggyCreatures.FusionCache/FusionCacheEntryOptions.cs [hunk 1]
```diff
@@ -63,6 +65,16 @@ public FusionCacheEntryOptions(TimeSpan? duration = null)
 	/// </summary>
 	public TimeSpan Duration { get; set; }
 
+	/// <summary>
+	///  Gets the Microsoft.Extensions.Primitives.IChangeToken instances which cause the cache entry to expire.
```
Comment (tacosontitan) [inline src/ZiggyCreatures.FusionCache/FusionCacheEntryOptions.cs:69]: Utilizing a `<see cref="..."/>` tag here will enable consumers to navigate to the specified type.

File: src/ZiggyCreatures.FusionCache/FusionCacheEntryOptions.cs [hunk 2]
```diff
@@ -63,6 +65,16 @@ public FusionCacheEntryOptions(TimeSpan? duration = null)
...
+	/// <br/><br/>	
+	/// Cancelling th source forces an instant eviction from cache and calls FusionCache.Remove on the key to remove from distributed cache and send backplane anotification
+	/// <br/><br/>	
+	/// If set on DefaultEntryOptions, reinstantiate this property every time the CancellationTokenSource is Cancelled.
```
Comment (tacosontitan) [inline src/ZiggyCreatures.FusionCache/FusionCacheEntryOptions.cs:73]: Recommending the use of `see` with `DefaultEntryOptions` and `CancellationTokenSource` here. Additionally, not sure if `Cancelled` is supposed to be capitalized.

**Comment by jodydonetti** [comment]
Posted my thinking [here](https://github.com/ZiggyCreatures/FusionCache/discussions/61#discussioncomment-5159731).
Closing for now since there has not been a response.
Will reopen for sure in case something new will came up.

**Pull request closed without merging**